### PR TITLE
adding documentation for in-browser testing using nightwatch

### PR DESCRIPTION
**Pull request by spriggsha**
Adding to dev documentation.

### CODE REVIEW

File: docs/developer-guide/browser-testing.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  "devDependencies": {
+    "bower": "^1.3.12",
+	...
+    "nightwatch": "^0.5.36"
+}</code>
+
+##Get set up with Sauce Labs
+Since Next is an open-source project, we're able to use Sauce Labs for free, which saves us the hassle of maintaining our own, local Selenium Grid.
+
+Go to the [signup page](https://saucelabs.com/opensauce) for open source projects and create an account.  There are two things you're going to need when you're done:
```
Comment (adgad) [inline docs/developer-guide/browser-testing.md]: Do we need individual accounts here? Or 'app' accounts (e.g. ft-next-dobi)? Or is there a 'next' account...

**Comment by adgad** [comment]
:+1: 

Probably worth at some point considering being able to run locally (i.e. localhost:5050) - which will make it easier to be a bit more test-driven rather than having to check in/deploy before running.

**Comment by matthew-andrews** [comment]
@adgad part of the work to put all the config vars onto a service will make it easier to spin up new apps on pull request.   But I think it runs locally too.

File: docs/developer-guide/browser-testing.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+---
+
+# Browser testing
+Next uses [Nightwatch](http://nightwatchjs.org/) to run our in-browser tests.  Nightwatch is a testing framework built around Selenium which can run tests either against any server that implements Selenium Grid software (including [Sauce Labs](https://saucelabs.com)).
+
+The Nightwatch developer guide can be found [here](http://nightwatchjs.org/guide#guidehttp://nightwatchjs.org/guide#guide) and the API reference is [here](http://nightwatchjs.org/api#assertions).  You can also use the implementation at [engels](https://github.com/Financial-Times/next-engels/tree/master/tests) as a guide.   
+
+##Add nightwatch to your dev dependencies
+
+<code>},
```
Comment (matthew-andrews) [inline docs/developer-guide/browser-testing.md]: This can be simplified to:-
```
npm install --save-dev nightwatch
```

Comment (matthew-andrews) [inline docs/developer-guide/browser-testing.md]: (which installs the dependency and adds it to the `package.json` in a single command)

File: docs/developer-guide/browser-testing.md (hunk 1 above)
Comment (matthew-andrews) [inline docs/developer-guide/browser-testing.md]: afaik one account per sauce labs

**Pull request merged**